src/Symfony/Component/DependencyInjection/Alias.php

```diff
@@ -25,6 +25,7 @@ public function __construct($id, $public = true)
     {
         $this->id = (string) $id;
         $this->public = $public;
+        $this->private = 2 > func_num_args();
```

Comment:
shouldn't this be `$public && 2 > func_num_args()` ?

If I register an alias as private explicitly, it does not need to have the BC layer keeping it from being inlined.

Reply:
when "private" is set to true, the BC layer is enabled and triggers a deprecation,
so here, we can it to be "false" when either true or false is passes.

Reply:
hmm sorry, I misunderstood the condition (reading too fast).